packages/react/src/themes/teams/components/Alert/alertStyles.ts

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react'
+import {
+  ComponentSlotStylesInput,
+  ICSSInJSStyle,
+  SiteVariablesPrepared,
+  NaturalColors,
+} from '../../../types'
+import { AlertProps } from '../../../../components/Alert/Alert'
+import { AlertVariables } from './alertVariables'
+
+const getIntentColorsFromProps = (
+  p: AlertProps,
+  v: AlertVariables,
+  siteVars: SiteVariablesPrepared,
+): React.CSSProperties => {
+  const { colors } = siteVars
+  const naturalColors: NaturalColors = siteVars.naturalColors
+
+  if (p.danger) {
+    return {
+      color: siteVars.red,
+      backgroundColor: siteVars.red10,
+      borderColor: siteVars.red08,
+    }
+  }
+
+  if (p.info) {
+    return {
+      color: colors.grey[900],
+      backgroundColor: siteVars.gray09,
+      borderColor: siteVars.gray08,
+    }
+  }
+
+  if (p.success) {
+    return {
+      color: colors.green[900], // $app-green-04
+      backgroundColor: colors.grey[50], // $app-white
+      borderColor: naturalColors.lightGreen[900], // $app-green
+    }
+  }
+
+  if (p.warning) {
+    return {
+      color: siteVars.gray03,
+      backgroundColor: colors.grey[50], // $app-white
+      borderColor: colors.yellow[900], // $app-yellow
+    }
+  }
+
+  return {
+    color: v.color,
+    backgroundColor: v.backgroundColor,
+    borderColor: v.borderColor,
+  }
+}
+
+const alertStyles: ComponentSlotStylesInput<AlertProps, AlertVariables> = {
+  root: ({ props: p, variables: v, theme: { siteVariables } }): ICSSInJSStyle => {
+    return {
+      display: 'flex',
```

Comment:
Is there a reason why we are not using the Flex component as a root of the Alert? We are adding some styles that are already there.

Reply:
I was considering it and decided to avoid for 2 reasons:

### 1. Markup becomes complex

```typescript
renderComponent(config: RenderResultConfig<AlertProps>) {
  const { accessibility, classes, ElementType, unhandledProps, styles } = config
  const { children, content, onClose, closeButton } = this.props
  const hasChildren = childrenExist(children)
  return (
    <Flex
      as={ElementType}
      gap="gap.smaller"
      vAlign="center"
      className={classes.root}
      {...accessibility.attributes.root}
      {...rtlTextContainer.getAttributes({ forElements: [children, content] })}
      {...unhandledProps}
    >
      {hasChildren && children}
      {!hasChildren && [
        <Flex.Item grow>
          {Box.create(content, {
            defaultProps: {
              key: Alert.slotClassNames.content,
              className: Alert.slotClassNames.content,
              styles: styles.content,
            },
          })}
        </Flex.Item>,
        onClose &&
          Button.create(closeButton, {
            defaultProps: {
              key: Alert.slotClassNames.closeButton,
              iconOnly: true,
              text: true,
              className: Alert.slotClassNames.closeButton,
              styles: styles.closeButton,
            },
            overrideProps: this.handleCloseButtonOverrides,
          }),
      ]}
    </Flex>
  )
}
```

### 2. We're tying the `Alert` implementation with the flex display which is more a theme concern

Other themes might not want to have a flex display. It is true they could override this with styles but it can become messy

What do you think, @mnajdova ?

Reply:
agreed to not use `Flex`